### ISSUE TITLE
False positive for `no-render-in-setup`

### ISSUE THREAD

**Issue by StefanoMagrassi** [testing-library/eslint-plugin-testing-library#505]
### Plugin version

5.0.0

### ESLint version

8.2.0

### Node.js version

14.18.1

### npm/yarn version

8.1.3

### Operating system

Ubuntu 20.04

### Bug description

Using any function or method which starts with `render` in `beforeAll`/`beforeEach` is flagged as an error, even if it's not related with RTL's `render` function

### Steps to reproduce

```jsx
const renderSpy = jest.fn();

beforeAll(() => {
  renderSpy();  // Forbidden usage of `render` within testing framework `beforeAll` setup
});
```

### ESLint configuration
```js
module.exports = {
  root: true,

  plugins: ['react', 'testing-library'],

  extends: [
    'plugin:react/recommended',
    'plugin:react-hooks/recommended',
    'plugin:testing-library/react',
    'contactlab/typescript',
    'prettier'
  ],

  parserOptions: {
    project: 'tsconfig.eslint.json'
  },

  env: {
    jest: true
  },

  rules: {
    // --- ES6
    'no-duplicate-imports': 'error',
    camelcase: [ 'error',],

    // --- Typescript
    '@typescript-eslint/ban-types': [
      'error',
      {
        types: {
          '{}': false
        },
        extendDefaults: true
      }
    ],
    '@typescript-eslint/no-unused-vars': [
      'error',
      {
        argsIgnorePattern: '^_'
      }
    ],

    // --- React
    'react/react-in-jsx-scope': 'off',
    'react/jsx-uses-vars': 'error',
    'react/prop-types': 'off',
    'react/display-name': 'off',

    // --- JSDoc
    'jsdoc/no-multi-asterisks': 'off'
  },

  overrides: [
    // --- Disable typescript rules for tests files
    {
      files: ['*.spec.ts*', 'testing/**/*.ts'],
      rules: {
        '@typescript-eslint/explicit-function-return-type': 'off',
        '@typescript-eslint/no-explicit-any': 'off'
      }
    },
    {
      files: ['e2e/**/*.ts'],
      globals: {
        fixture: 'readonly'
      },
      rules: {
        '@typescript-eslint/no-explicit-any': 'off'
      }
    }
  ],

  settings: {
    react: {
      version: 'detect'
    }
  }
};
```

### Rule(s) affected

no-render-in-setup

### Do you want to submit a pull request to fix this bug?

No

**Comment by Belco90** [comment]
Hi @StefanoMagrassi! This is related to Aggressive Reporting. Have you checked [the Troubleshooting section](https://github.com/testing-library/eslint-plugin-testing-library#troubleshooting) to find more about restricting this behavior?

**Comment by StefanoMagrassi** [comment]
@Belco90 thank you, I didn't aware of "Aggressive Renders Reporting": I'll switch off this feature, but I think that this kind of logic

> will assume that all methods which names contain "render" should be reported

it's too much "aggressive" in general.

**Comment by Belco90** [comment]
It might be, but we have proper settings to tweak it as needed. Otherwise, it could silent other potential errors that no one would notice.
